### PR TITLE
refactor(frontend): extract task approval transition resolver

### PR DESCRIPTION
**Pull request by Maxsky5**
Summary: Extracts task approval transition decisions out of the large Kanban approval hook into focused pure helpers with dedicated coverage.

Goal:
This PR reduces the complexity of `useTaskApprovalFlow`, where approval, rejection, missing-builder-worktree recovery, pull request completion, direct merge completion, and conflict handling were all interleaved in one hook. The approval path is business-critical because it decides whether Beads tasks close, return to builder rework, recover from missing worktrees, or continue through pull request/direct merge completion. Pulling those decisions into smaller modules makes the state rules easier to review and safer to extend.

Technical choices:
- Added `task-approval-transition-resolver.ts` for pure status/action/mode/submission routing decisions. It uses narrow task shapes and Beads task actions instead of coupling tests to full task cards.
- Added table-driven resolver tests for approve, request changes, blocked/in-progress unavailable states, human review, AI review, PR-linked completion, default mode selection, and submission routing.
- Extracted approval modal state construction and git-conflict orchestration into focused helpers so `useTaskApprovalFlow` keeps orchestration responsibilities without carrying all decision logic inline.
- Kept the implementation frontend-scoped and contract-aligned; no fallback behavior or Rust/backend API changes were introduced.

Verification:
- `rtk bun run lint`
- `rtk bun run typecheck`
- `rtk bun run test`
- `rtk bun run build`
- `rtk cargo fmt --all --check` from `apps/desktop/src-tauri`
- `rtk cargo clippy --workspace --all-targets -- -D warnings` from `apps/desktop/src-tauri`
- `rtk cargo check --workspace` from `apps/desktop/src-tauri`
- `rtk cargo test --workspace` from `apps/desktop/src-tauri`
- `rtk cargo build --workspace` from `apps/desktop/src-tauri`

## Summary by CodeRabbit

* **Refactor**
  * Reorganized task approval workflow logic, centralizing modal and submission-flow management for improved maintainability.

* **New Features**
  * Added dedicated Git conflict resolution dialog to handle conflicts during task approval process.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Maxsky5/openducktor/pull/650?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/frontend/src/pages/kanban/use-task-approval-flow.ts [hunk 1]
```diff
@@ -1,39 +1,34 @@
 import type { TaskApprovalContextLoadResult, TaskCard } from "@openducktor/contracts";
 import { useQueryClient } from "@tanstack/react-query";
-import { useCallback, useReducer, useRef, useState } from "react";
+import { useCallback, useReducer, useRef } from "react";
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/use-task-approval-flow.ts:3]: Not changing this one: the hook return is immediately destructured by the caller, so memoizing the wrapper object would not stabilize a consumed dependency and would add churn without a measured issue.

File: packages/frontend/src/pages/kanban/use-task-approval-flow.ts [hunk 2]
```diff
@@ -45,18 +40,16 @@ type UseTaskApprovalFlowArgs = {
...
-  open: false,
-  taskId: null,
-  conflict: null,
-  isHandlingConflict: false,
-  conflictAction: null,
+type TaskApprovalOpenOptions = {
+  mode?: "direct_merge" | "pull_request";
+  pullRequestDraftMode?: "manual" | "generate_ai";
+  errorMessage?: string | null;
+};
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/use-task-approval-flow.ts]: Fixed in 27f341dca: moved TaskApprovalOpenOptions to the shared kanban page model types and reused it from useTaskApprovalFlow.

File: packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
+import { useCallback, useState } from "react";
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts:1]: Not changing this one: the caller immediately destructures this hook result, so memoizing the returned wrapper object would not improve referential stability for the consumed values.

File: packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+import { toast } from "sonner";
+import type { GitConflict, GitConflictAction } from "@/features/agent-studio-git";
+import { getGitConflictCopy } from "@/features/git-conflict-resolution";
+import { errorMessage } from "@/lib/errors";
+import type { KanbanPageModels } from "./kanban-page-model-types";
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts]: Fixed in 27f341dca: the git-conflict hook now imports the shared TaskApprovalOpenOptions type from the kanban page model types.

File: packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+} from "./task-approval-flow-git-conflict";
+
+type OpenTaskApproval = (
+  taskId: string,
+  options?: {
+    mode?: "direct_merge" | "pull_request";
+    pullRequestDraftMode?: "manual" | "generate_ai";
+    errorMessage?: string | null;
+  },
+) => void;
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts]: Fixed in 27f341dca: OpenTaskApproval now uses the shared TaskApprovalOpenOptions type instead of duplicating the options shape inline.

File: packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        },
+        onAbort: abortGitConflict,
+        onAskBuilder: askBuilderToResolveGitConflict,
+      }
+    : null;
+
+  return {
+    taskGitConflictDialog,
+    openGitConflictDialog,
+  };
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts:162]: Not changing this one: the hook result is immediately destructured where it is consumed, so memoizing this wrapper object would add churn without affecting the actual dependencies used by callers.

File: packages/frontend/src/pages/kanban/task-approval-transition-resolver.ts [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    }
+  | { kind: "submit_direct_merge"; approval: TaskApprovalFlowReadyState; repoPath: string }
+  | { kind: "submit_pull_request"; approval: TaskApprovalFlowReadyState; repoPath: string };
+
+const REVIEW_STATUSES = new Set<TaskStatus>(["ai_review", "human_review"]);
+
+const isReviewStatus = (status: TaskStatus): status is ReviewStatus => REVIEW_STATUSES.has(status);
+
+const hasTaskAction = (task: Pick<TaskCard, "availableActions">, action: TaskAction): boolean =>
+  task.availableActions.includes(action);
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/task-approval-transition-resolver.ts:54]: Not changing this one: availableActions is contract-normalized into a required array. Adding optional chaining here would hide a broken parsed task contract, which conflicts with the project no-fallback guidance.

File: packages/frontend/src/pages/kanban/task-approval-modal-model.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+export const buildTaskApprovalModalModel = ({
+  completeDirectMerge,
+  confirm,
+  dispatch,
+  reset,
+  resetMissingBuilderWorktree,
+  state,
+}: BuildTaskApprovalModalModelArgs): TaskApprovalModalModel => {
+  const approvalContext = state.approvalContext;
+  const githubProvider = approvalContext?.providers.find((entry) => entry.providerId === "github");
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/task-approval-modal-model.ts:28]: Not changing this one: providers is part of the parsed approval-context contract and is normalized as an array. Optional fallback here would mask malformed contract data instead of surfacing the source issue.

File: packages/frontend/src/pages/kanban/task-approval-transition-resolver.test.ts [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+    const route = resolveTaskApprovalSubmissionRoute(state, null);
+
+    expect(route.kind).toBe("complete_missing_builder_worktree");
+    if (route.kind !== "complete_missing_builder_worktree") {
+      throw new Error(`Expected missing-builder route, received ${route.kind}`);
+    }
+
+    expect(route.approval.stage).toBe("missing_builder_worktree");
+    expect(route.approval.taskId).toBe(state.taskId);
+  });
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/task-approval-transition-resolver.test.ts:333]: Fixed in 27f341dca: added explicit action_unavailable coverage and related resolver edge cases for request_changes, ai_review, missing task data, unavailable approval transitions, and direct-merge completion submission state.

File: packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts]: Fixed in 27f341dca: TaskApprovalOpenOptions now lives in the shared kanban page model types and both hooks use that single definition.

**Comment by Maxsky5** [comment]
Addressed the substantive review feedback in 27f341dca.

Fixed:
- Added resolver coverage for action_unavailable, request_changes on non-review status, request_changes from ai_review, undefined task data, unavailable approval transitions with PR context, and direct-merge completion submission state.
- Moved TaskApprovalOpenOptions into the shared kanban page model types and reused it from both hooks.
- Added an explicit ignore route for complete_direct_merge in resolveTaskApprovalSubmissionRoute.

Kept unchanged:
- Did not add PR state matrix tests: this resolver only branches on linked PR presence; PR state semantics belong to provider/metadata handling.
- Did not change non-reviewable task + PR defaulting: approval workflow availability remains the gate before forcing PR mode, and the new test documents the current context/default behavior.
- Did not add a workspaceRepoPath guard to Ask Builder: abort needs the repo path, while Ask Builder does not use it.
- Did not memoize hook return wrapper objects: callers immediately destructure the hook results, so memoizing the wrapper would not stabilize consumed dependencies.
- Did not add optional chaining fallbacks for availableActions/providers: both are contract-normalized arrays, and fallback logic would mask malformed parsed contract data.

Verification passed locally: bun run lint, bun run typecheck, bun run test, bun run build, cargo fmt --all --check, cargo clippy --workspace --all-targets -- -D warnings, cargo check --workspace, cargo test --workspace, cargo build --workspace.

File: packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+          description,
+        });
+        setGitConflictState((current) => ({
+          ...current,
+          isHandlingConflict: false,
+          conflictAction: null,
+        }));
+      }
+    })();
+  }, [closeGitConflict, gitConflictState, onResolveGitConflict, reset]);
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/frontend/src/pages/kanban/use-task-approval-git-conflict-flow.ts:141]: Fixed in 2bc8706f1: the task approval git-conflict hook now uses getGitConflictCopy(conflict.operation).builderFailureMessage for the Ask Builder failure toast, with the existing error detail as the description.
